Library/ViewModels/ManagePledgeSummaryViewModel.swift

```diff
@@ -0,0 +1,133 @@
+import Foundation
+import KsApi
+import Prelude
+import ReactiveSwift
+
+public protocol ManagePledgeSummaryViewModelInputs {
+  func configureWith(_ project: Project)
+}
+
+public protocol ManagePledgeSummaryViewModelOutputs {
+  var backerNumberText: Signal<String, Never> { get }
+  var backingDateText: Signal<String, Never> { get }
+  var pledgeAmountText: Signal<NSAttributedString, Never> { get }
+  var shippingAmountText: Signal<NSAttributedString, Never> { get }
+  var shippingLocationStackViewIsHidden: Signal<Bool, Never> { get }
+  var shippingLocationText: Signal<String, Never> { get }
+  var totalAmountText: Signal<NSAttributedString, Never> { get }
+}
+
+public protocol ManagePledgeSummaryViewModelType {
+  var inputs: ManagePledgeSummaryViewModelInputs { get }
+  var outputs: ManagePledgeSummaryViewModelOutputs { get }
+}
+
+public class ManagePledgeSummaryViewModel: ManagePledgeSummaryViewModelType,
+  ManagePledgeSummaryViewModelInputs, ManagePledgeSummaryViewModelOutputs {
+  public init() {
+    let backing = self.projectSignal
+      .map { $0.personalization.backing }
+      .skipNil()
+
+    let projectAndBacking = self.projectSignal
+      .zip(with: backing)
+
+    self.backerNumberText = backing
+      .map { Strings.backer_modal_backer_number(backer_number: Format.wholeNumber($0.sequence)) }
+
+    self.backingDateText = backing
+      .map(formattedPledgeDate)
+
+    self.pledgeAmountText = projectAndBacking
+      .map { attributedCurrency(with: $0.0, amount: $0.1.amount) }
+      .skipNil()
+
+    let shippingAmount = backing
+      .map { Double($0.shippingAmount ?? 0) }
+
+    self.shippingAmountText = self.projectSignal
+      .combineLatest(with: shippingAmount)
+      .map { shippingValue(with: $0.0, with: $0.1) }
+      .skipNil()
+
+    self.totalAmountText = projectAndBacking
+      .combineLatest(with: shippingAmount)
+      .map(unpack)
+      .map { project, backing, shippingAmount in
+        attributedCurrency(with: project, amount: backing.amount + shippingAmount)
+      }
+      .skipNil()
+
+    self.shippingLocationText = backing.ignoreValues()
+      .map { Strings.Shipping() + ": " + "Australia" }
```

Comment:
Here the `Australia` is hardcoded because we will use a Backing property called `locationName`. This is done in another branch (for the 2/2 PR)

Reply:
Ok, I think we want to add a new string here:

`"Shipping: %{country_name}"`

Although it seems very much like what you have, by including the `:` in the string we give the translator the ability to move it around if they need to.

Reply:
Yeah, before I implemented this, I checked out the Android codebase and that's how it was implemented. I can create a new string on the second part of this work, but I'm not sure if it would make a big difference.
cc @ginarovirosa

Reply:
hello @Scollaco @justinswart. Indeed having the `Shipping: %{country_name}` format makes it easier for the translators to work around country name prefixes.  Fwiw, yt does seem that the `Shipping: %{country_name}` variable is an old one. We use `%{country}` now. As you can see in this string for instance: `en -> pledges -> checkout_summary -> Shipping_to_country `

Reply:
Thanks @ginarovirosa!